homeassistant/components/mqtt/client.py

```diff
@@ -719,7 +719,14 @@ def _mqtt_handle_message(self, msg: mqtt.MQTTMessage) -> None:
                     timestamp,
                 ),
             )
-        self._mqtt_data.state_write_requests.process_write_state_requests()
+        # pylint: disable-next=expression-not-assigned
+        catch_log_exception(
+            self._mqtt_data.state_write_requests.process_write_state_requests,
+            lambda msg: (
+                "Exception on handling write state requests when handling msg on "
+                f"'{msg.topic}': '{msg.payload}'"
+            ),
+        )()
```

Comment:
It was my mistake to suggest catch_log_exception, and pylint is rightly complaining; this is wasteful because we create a new function every time we call `process_write_state_requests`

We should do something like this instead:

```suggestion
        try:
            self._mqtt_data.state_write_requests.process_write_state_requests()
        except Exception:  # pylint: disable=broad-except
           _LOGGER.error("blah", exc_info=True)
```